internal/services/databricks/databricks_workspace_resource.go

```diff
@@ -277,32 +279,51 @@ func resourceDatabricksWorkspace() *pluginsdk.Resource {
 			},
 
 			"managed_services_cmk_key_vault_key_id": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ValidateFunc: keyVaultValidate.KeyVaultChildID,
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				ValidateFunc:  keyVaultValidate.KeyVaultChildID,
+				ConflictsWith: []string{"managed_services_cmk_managed_hsm_key_id"},
+			},
+
+			"managed_services_cmk_managed_hsm_key_id": {
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				ValidateFunc:  hsmValidate.ManagedHSMDataPlaneVersionedKeyID,
+				ConflictsWith: []string{"managed_services_cmk_key_vault_id"},
 			},
 
 			"managed_services_cmk_key_vault_id": {
 				Type:         pluginsdk.TypeString,
 				Optional:     true,
 				ValidateFunc: commonids.ValidateKeyVaultID,
+				Deprecated:   "Use 'managed_services_cmk_key_vault_key_id' instead, this field is not needed for cross subscription key vaults anymore",
 			},
 
 			"managed_disk_cmk_key_vault_key_id": {
-				Type:         pluginsdk.TypeString,
-				Optional:     true,
-				ValidateFunc: keyVaultValidate.KeyVaultChildID,
+				Type:          pluginsdk.TypeString,
+				Optional:      true,
+				ValidateFunc:  keyVaultValidate.KeyVaultChildID,
+				ConflictsWith: []string{"managed_disk_cmk_managed_hsm_key_id"},
```

Comment:
Shouldn't this also conflict with `managed_disk_cmk_key_vault_id`?
```suggestion
				ConflictsWith: []string{"managed_disk_cmk_managed_hsm_key_id", "managed_disk_cmk_key_vault_id"},
```